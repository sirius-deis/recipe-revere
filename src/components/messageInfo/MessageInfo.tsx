import { FC } from "react";
import styles from './MessageInfo.module.css'
import { IoCheckmarkDoneOutline } from "react-icons/io5";
import { IoCheckmarkOutline } from "react-icons/io5";
import { IMessageInfo } from "../../types/types";

const formatTime = (timestamp: number): string => {
  const date = new Date(timestamp);
  const hours = date.getHours().toString().padStart(2, '0');
  const minutes = date.getMinutes().toString().padStart(2, '0');
  return `${hours}:${minutes}`;
}

const MessageInfo: FC<IMessageInfo> = ({ timestamp, isRead, isSend }) => {
  return (
    <div className={styles['message-info']}>
      <span className={styles.timestamp}>{formatTime(timestamp)}</span>
      <span className={styles['read-status']}>
        {isRead && <IoCheckmarkDoneOutline data-testid="read-icon" />}
        {!isRead && isSend && <IoCheckmarkOutline data-testid="send-icon" />}
      </span>
    </div>
  )
}

export default MessageInfo;