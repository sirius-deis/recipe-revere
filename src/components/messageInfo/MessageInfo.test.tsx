import { render, screen } from "@testing-library/react";
import MessageInfo from "./MessageInfo";

describe("MessageInfo component", () => {
  it("should match snapshot", () => {
    const { container } = render(<MessageInfo timestamp={1631360000} isRead={true} isSend={true} />);
    expect(container).toMatchSnapshot();
  });

  it("should render read status icon", () => {
    render(<MessageInfo timestamp={1631360000} isRead={true} isSend={true} />);
    const readIcon = screen.getByTestId('read-icon');
    expect(readIcon).toBeInTheDocument();
    const sendIcon = screen.queryByTestId('send-icon');
    expect(sendIcon).not.toBeInTheDocument();
  });

  it("should render send status icon", () => {
    render(<MessageInfo timestamp={1631360000} isRead={false} isSend={true} />);
    const sendIcon = screen.getByTestId('send-icon');
    expect(sendIcon).toBeInTheDocument();
    const readIcon = screen.queryByTestId('read-icon');
    expect(readIcon).not.toBeInTheDocument();
  });

  it("should render timestamp", () => {
    render(<MessageInfo timestamp={new Date("2024-07-19T12:24:04").getTime()} isRead={true} isSend={false} />);
    const timestamp = screen.getByText('12:24');
    expect(timestamp).toBeInTheDocument();
  });
})