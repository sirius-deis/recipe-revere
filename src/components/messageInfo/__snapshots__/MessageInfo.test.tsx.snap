// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInfo component should match snapshot 1`] = `
<div>
  <div
    class="message-info"
  >
    <span
      class="timestamp"
    >
      22:09
    </span>
    <span
      class="read-status"
    >
      <svg
        data-testid="read-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 128 240 384l-96-96m0 96-96-96m320-160L232 284"
          fill="none"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="32"
        />
      </svg>
    </span>
  </div>
</div>
`;
